datamade/bga-payroll
# Manual control of third-party search indexing

### This PR:

- Adds a `noindex` field to the Person model
- Enables the `Person` admin module
  - People are searchable by name
  - People are displayed with their most recent job
  - `noindex` can be toggled on and off from the change form, but no other edits may be made
- Updates the application code to use `noindex`

### Convenience changes:

- Dynamically configure the Redis port (`from django.conf import settings` instead of importing from the `local_settings` file directly), so the test config value is used when running the tests

### Testing instructions:

- Run the app locally, [per the instructions in the README](https://github.com/datamade/bga-payroll/blob/master/README.md).
  - Alternate course: Coordinate with me, and you can test on my local setup.
- Navigate to the admin page and, at the bottom, click "People". Verify that a list of people and their titles appear.
- Search a Person by name in the search bar. Verify that the results make sense, e.g., searching for Joe returns Joe, Joella, etc. (N.b., we are _not_ supporting fuzzy match here.)
- Click a Person from the list. Verify that you are taken to a form that displays name, job, and slug, and allows you to check a box for `noindex`. Copy the slug to your clipboard.
- Check `noindex`, and click Save. Verify that you are returned to the listing page, with a confirmation message at the top.
- Navigate to `localhost:8000/person/YOUR-SLUG`. View the page source, and confirm that `<meta name="robots" content="noindex">` appears.
